validate/validate_test.go

```diff
@@ -557,6 +557,26 @@ func TestCheckLinux(t *testing.T) {
 			},
 			expected: specerror.ReadonlyPathsAbs,
 		},
+		{
+			val: rspec.Spec{
+				Version: "1.0.0",
+				Linux: &rspec.Linux{
+					Resources: &rspec.LinuxResources{
+						BlockIO: &rspec.LinuxBlockIO{
+							WeightDevice: []rspec.LinuxWeightDevice{
+								{
+									{
```

Comment:
From [Travis][1]:

```
validate/validate_test.go:568:10: missing type in composite literal
validate/validate_test.go:568:10: implicit assignment of unexported field 'linuxBlockIODevice' in specs.LinuxWeightDevice literal
```

[1]: https://travis-ci.org/opencontainers/runtime-tools/jobs/337923884#L556

Reply:
I don't know what to do with this, but I tried the following calls, and they all went wrong.
```
 WeightDevice: []rspec.LinuxWeightDevice{
        linuxBlockIODevice{
                Major: 5,
                Minor: 0,
        },
}
```
```
 WeightDevice: []rspec.LinuxWeightDevice{
        rspec.linuxBlockIODevice{
                Major: 5,
                Minor: 0,
        },
}
```

Reply:
> I don't know what to do with this, but I tried the following calls, and they all went wrong.

I'm pretty sure you want:

```go
WeightDevice: []rspec.LinuxWeightDevice{
  rspec.LinuxWeightDevice{
    Major: 5,
    Minor: 0,
  },
},
```

Reply:
This is still wrong.

Reply:
> This is still wrong.

Hmm.  Maybe Go has a problem with literal struct initialization for lowercase embedded types.  We may need to skip these for the literal initialization and add them later via the promoted properties:

```go
weightDevices := []rspec.LinuxWeightDevice{
  rspec.LinuxWeightDevice{},
}
weightDevices[0].Major = 5
weightDevices[0].Minor = 0
```

Then use `WeightDevice: weightDevices` when you initialize your config struct.

Reply:
Cool. It worked, thanks.